src/context.rs

```diff
@@ -379,18 +380,18 @@ impl<'buf> Secp256k1<SignOnlyPreallocated<'buf>> {
     #[inline]
     pub fn preallocate_signing_size() -> usize { Self::preallocate_size_gen() }
 
-    /// Creates a context from a raw context.
+    /// Creates a context from a raw context that can only be used for signing.
     ///
     /// # Safety
     ///
-    /// This is highly unsafe, due to the number of conditions that aren't checked.
-    /// * `raw_ctx` needs to be a valid Secp256k1 context pointer.
-    /// that was generated by *exactly* the same code/version of the libsecp256k1 used here.
-    /// * The capabilities (All/SignOnly/VerifyOnly) of the context *must* match the flags passed to libsecp256k1
-    /// when generating the context.
-    /// * The user must handle the freeing of the context(using the correct functions) by himself.
-    /// * This list *is not* exhaustive, and any violation may lead to Undefined Behavior.
+    /// This is highly unsafe due to the number of conditions that aren't checked, specifically:
     ///
+    /// * `raw_ctx` needs to be a valid Secp256k1 context pointer that was generated by *exactly*
+    ///   the same code/version of the libsecp256k1 used here.
+    /// * The capabilities (`All`/`SignOnly`/`VerifyOnly`) of the context *must* match the flags
+    ///   passed to libsecp256k1 when generating the context.
+    /// * The user must handle the freeing of the context (using the correct functions) by himself.
+    /// * `raw_ctx` must point to writable memory (cannot be `ffi::secp256k1_context_no_precomp`).
```

Comment:
The lifetime of `raw_ctx` pointer must outlive `'buf`.

Non-safety warning: the returned value will **not** deallocate the pointer, nor destroy the context which may lead to memory leaks. The caller is responsible for these. Note that `ManuallyDrop::drop` (or `drop_in_place`) will **only** destroy the context, it will **not** free memory behind the pointer.

Reply:
Are you trolling me with your cutnpasta? :) (I'm joking, thanks for making sure I don't only fix one instance of the docs)